components/block/element.js

```diff
@@ -9,62 +10,112 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    loading: {type: Boolean, state: true},
+    blockId: {type: String},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
+  constructor() {
+    super();
+
+    this.loading = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.model.on('sidebar_save:success', this.refresh.bind(this));
   }
 
-  edit(e){
-    console.log(e.target.innerText)
-    this.refresh();
+  get layout() {
+    return window.parent.Core.g.layout;
   }
 
-  delete(e) {
-    console.log(e.target.innerText)
-    this.remove();
+  get model() {
+    return (this.cached_model ||= this.layout.blocks.findWhere({
+      id: this.blockId,
+    }));
   }
 
-  moveUp(e) {
-    console.log(e.target.innerText)
-    const prevElement = this.previousElementSibling;
-    isBlock(prevElement) && this.parentNode.insertBefore(this, prevElement)
+  get parent() {
+    if (this.cached_parent !== undefined) return this.cached_parent;
+
+    const parentId = this.model.attributes.parent_block_id;
+    const parent = document.querySelector(`ngl-block[blockId="${parentId}"]`);
+
+    if (parent) return (this.cached_parent = parent);
+    else return (this.cached_parent = null);
   }
 
-  moveDown(e) {
-    console.log(e.target.innerText)
-    const nextElement = this.nextElementSibling;
-    isBlock(nextElement) && this.parentNode.insertBefore(nextElement, this)
+  get isInLinkedZone() {
+    return this.model.zone().is_linked();
+  }
+
+  renderMenu() {
+    if (this.isInLinkedZone) return this.renderLinkedBlockMenu();
+    else if (this.parent) return this.renderInnerBlockMenu();
+    else return this.renderOuterBlockMenu();
```

Comment:
Because you are using `return` you can also simplify this to:

```js
 if (this.isInLinkedZone) return this.renderLinkedBlockMenu();
 if (this.parent) return this.renderInnerBlockMenu();

 return this.renderOuterBlockMenu();
```

There is no point of `else return`

Reply:
Removed in f4602e9bd9225f12b6fb88d62b34ba95b82b62eb